`pnpm i` followed by `pnpm i --frozen-lockfile` complains about outdated lockfile

We have managed to land in a state where `pnpm i --frozen-lockfile` compains that pnpm-lock.yaml is not up to date, but `pnpm i` does not fix it. This issue appeared since the latest release.

### pnpm version:

v7.6.0

### Code to reproduce the issue:

```
git clone https://github.com/remotion-dev/remotion remotion-pnpm
cd remotion-pnpm
git checkout f76e74cec19d80348813760ca053a4af83ab71bc
pnpm i
pnpm i --frozen-lockfile
```

### Expected behavior:

No error

### Actual behavior:

```
 ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up-to-date with packages/core/package.json
```

### Additional information:

Corepack is enabled
The original lockfile was generated by v7.2.1

 - `node -v` prints: v16.16.0
 - Windows, macOS, or Linux?: macOS

--- a/.changeset/breezy-geese-end.md
+++ b/.changeset/breezy-geese-end.md
@@ -1,0 +1,6 @@
+---
+"@pnpm/lockfile-utils": patch
+"pnpm": patch
+---
+
+pnpm should not consider a lockfile out-of-date if `auto-install-peers` is set to `true` and the peer dependency is in `devDependencies` or `optionalDependencies` [#5080](https://github.com/pnpm/pnpm/issues/5080).

--- a/packages/lockfile-utils/src/satisfiesPackageManifest.ts
+++ b/packages/lockfile-utils/src/satisfiesPackageManifest.ts
@@ -4,22 +4,23 @@ import {
   ProjectManifest,
 } from '@pnpm/types'
 import equals from 'ramda/src/equals.js'
+import omit from 'ramda/src/omit.js'
 
 export default (lockfile: Lockfile, pkg: ProjectManifest, importerId: string, opts?: { autoInstallPeers?: boolean }) => {
   const importer = lockfile.importers[importerId]
   if (!importer) return false
   let existingDeps = { ...pkg.devDependencies, ...pkg.dependencies, ...pkg.optionalDependencies }
   if (opts?.autoInstallPeers) {
-    existingDeps = {
-      ...pkg.peerDependencies,
-      ...existingDeps,
-    }
     pkg = {
       ...pkg,
       dependencies: {
-        ...pkg.peerDependencies,
+        ...omit(Object.keys(existingDeps), pkg.peerDependencies),
         ...pkg.dependencies,
       },
+    }
+    existingDeps = {
+      ...pkg.peerDependencies,
+      ...existingDeps,
     }
   }
   if (

--- a/packages/lockfile-utils/test/satisfiesPackageManifest.ts
+++ b/packages/lockfile-utils/test/satisfiesPackageManifest.ts
@@ -288,6 +288,36 @@ test('satisfiesPackageManifest()', () => {
     ...DEFAULT_LOCKFILE_FIELDS,
     importers: {
       '.': {
+        optionalDependencies: {
+          bar: '1.0.0',
+        },
+        devDependencies: {
+          foo: '1.0.0',
+        },
+        specifiers: {
+          foo: '1.0.0',
+          bar: '1.0.0',
+        },
+      },
+    },
+  }, {
+    ...DEFAULT_PKG_FIELDS,
+    optionalDependencies: {
+      bar: '1.0.0',
+    },
+    devDependencies: {
+      foo: '1.0.0',
+    },
+    peerDependencies: {
+      foo: '^1.0.0',
+      bar: '^1.0.0',
+    },
+  }, '.', { autoInstallPeers: true })).toBe(true)
+
+  expect(satisfiesPackageManifest({
+    ...DEFAULT_LOCKFILE_FIELDS,
+    importers: {
+      '.': {
         dependencies: {
           foo: '1.0.0',
         },

